test/e2e/latency.go

```diff
@@ -0,0 +1,268 @@
+				} else {
+					Failf("Pod %v is reported to be running, but none of its containers are", p.Name)
+				}
+			}
+		}
+	}
+
+	// Create a informer to read timestamps for each pod
+	stopCh := make(chan struct{})
+	_, informer := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func() (runtime.Object, error) {
+				return c.Pods(ns).List(labels.SelectorFromSet(labels.Set{"name": additionalPodsPrefix}), fields.Everything())
+			},
+			WatchFunc: func(rv string) (watch.Interface, error) {
+				return c.Pods(ns).Watch(labels.SelectorFromSet(labels.Set{"name": additionalPodsPrefix}), fields.Everything(), rv)
+			},
+		},
+		&api.Pod{},
+		0,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				p, ok := obj.(*api.Pod)
+				Expect(ok).To(Equal(true))
+				go readPodInfo(p)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				p, ok := newObj.(*api.Pod)
+				Expect(ok).To(Equal(true))
+				go readPodInfo(p)
+			},
+		},
+	)
+	go informer.Run(stopCh)
+
+	// Create  additional pods with throughput ~5 pods/sec.
+	var wg sync.WaitGroup
+	wg.Add(nodeCount)
+	podLabels := map[string]string{
+		"name": additionalPodsPrefix,
+	}
+	for i := 1; i <= nodeCount; i++ {
+		name := additionalPodsPrefix + "-" + strconv.Itoa(i)
+		go createRunningPod(&wg, c, name, ns, "gcr.io/google_containers/pause:go", podLabels)
+		time.Sleep(200 * time.Millisecond)
+	}
+	wg.Wait()
+
+	Logf("Waiting for all Pods begin observed by the watch...")
+	for start := time.Now(); len(watchTimestamps) < nodeCount; time.Sleep(10 * time.Second) {
+		if time.Since(start) < timeout {
+			Failf("Timeout reached waiting for all Pods being observed by the watch.")
+		}
+	}
+	close(stopCh)
+
+	// Read the schedule timestamp by checking the scheduler event for each pod
+	schedEvents, err := c.Events(ns).List(
+		labels.Everything(),
+		fields.Set{
+			"involvedObject.kind":      "Pod",
+			"involvedObject.namespace": ns,
+			"source":                   "scheduler",
+		}.AsSelector())
+	expectNoError(err)
+	for k := range createTimestamps {
+		for _, event := range schedEvents.Items {
+			if event.InvolvedObject.Name == k {
+				scheduleTimestamps[k] = event.FirstTimestamp
+				break
+			}
+		}
+	}
+
+	var (
+		scheduleLatencies        = make([]podLatencyData, 0)
+		startLatencies           = make([]podLatencyData, 0)
+		watchLatencies           = make([]podLatencyData, 0)
+		scheduleToWatchLatencies = make([]podLatencyData, 0)
+		e2eLatencies             = make([]podLatencyData, 0)
+	)
+
+	for name, podNode := range nodes {
+		createTs, ok := createTimestamps[name]
+		Expect(ok).To(Equal(true))
+		scheduleTs, ok := scheduleTimestamps[name]
+		Expect(ok).To(Equal(true))
+		runTs, ok := startTimestamps[name]
+		Expect(ok).To(Equal(true))
+		watchTs, ok := watchTimestamps[name]
+		Expect(ok).To(Equal(true))
+
+		scheduleLatencies = append(scheduleLatencies, podLatencyData{name, podNode, scheduleTs.Time.Sub(createTs.Time)})
```

Comment:
i had the actual latency measurements as vars because the inline style in this version is much much harder for me to read... can i get a pass on this one? sorry to pick nits after you already changed it back! @ashcrow :disappointed: